### PR TITLE
[spi_passthru] Test status-modifying opcodes 

### PR DESCRIPTION
**Pull request by cfrantz**
1. Test EN4B, EX4B, WREN and WRDI.
2. Test reading the extended status register.

Signed-off-by: Chris Frantz <cfrantz@google.com>

### CODE REVIEW

File: sw/device/tests/spi_passthru_test.c [hunk 1]
```diff
@@ -71,14 +83,9 @@ bool test_main(void) {
...
-  dif_spi_device_config_t dev_cfg = {
-      .clock_polarity = kDifSpiDeviceEdgePositive,
-      .data_phase = kDifSpiDeviceEdgeNegative,
-      .tx_order = kDifSpiDeviceBitOrderMsbToLsb,
-      .rx_order = kDifSpiDeviceBitOrderMsbToLsb,
-      .device_mode = kDifSpiDeviceModePassthrough,
-  };
-  CHECK_DIF_OK(dif_spi_device_configure(&spid, dev_cfg));
+  spi_device_testutils_configure_passthrough(&spid,
+                                             /*filters=*/0,
```
Comment (a-will) [inline sw/device/tests/spi_passthru_test.c:87]: Might want to filter everything (from reaching the downstream SPI flash), then selectively enable. Aren't the status register commands currently leaking through to the other end?

File: sw/host/tests/chip/spi_passthru/src/main.rs [hunk 2]
```diff
@@ -60,6 +64,71 @@ fn test_jedec_id(opts: &Opts, transport: &TransportWrapper) -> Result<()> {
...
+    let spi = transport.spi(&opts.spi)?;
+
+    let sr = StatusRegister {
+        status: 0x5A55AA,
+        addr_4b: false,
+    };
+    sr.write(&*uart)?;
+    // FIXME(cfrantz): Is this a bug?  We need a transaction to latch
+    // the staged flash_status into the register.
+    spi.run_transaction(&mut [Transfer::Write(&[SpiFlash::NOP])])?;
```
Comment (a-will) [inline sw/host/tests/chip/spi_passthru/src/main.rs:126]: A sharp edge? For initial software-provided values, the assumption is that Read ID or Read SFDP will execute first, allowing the staged value to commit.

If the status registers need to update independently of initial values or Write Status, then we have a bug. (WEL and WIP's special handling aside)

**Review by a-will — APPROVED** [review]

File: sw/device/tests/spi_passthru_test.c (hunk 1 above)
Comment (cfrantz) [reply sw/device/tests/spi_passthru_test.c]: Good catch.

File: sw/host/tests/chip/spi_passthru/src/main.rs (hunk 2 above)
Comment (cfrantz) [reply sw/host/tests/chip/spi_passthru/src/main.rs]: I think the current behavior is acceptable.  I left a comment explaining the behavior and the reason for the NOP transaction.

**Review by timothytrippel — APPROVED** [review]

**Pull request merged**